http-server/src/main/java/io/micronaut/http/server/exceptions/response/DefaultHtmlErrorResponseBodyProvider.java

```diff
@@ -0,0 +1,190 @@
+                                    text-rendering: optimizeLegibility;
+                                    -webkit-text-size-adjust: 100%;
+                                  }
+                                  a {
+                                    color: inherit;
+                                    font-weight: 700;
+                                    text-decoration: underline;
+                                    text-underline-offset: 0.0925em;
+                                  }            
+                                  b, strong {
+                                    font-weight: 700;
+                                  }            
+                                  i, em {
+                                    font-style: italic;
+                                  }            
+                                  main {
+                                    display: grid;
+                                    gap: 1em;
+                                    padding: 2em;
+                                    place-items: center;
+                                    text-align: center;
+                                  }            
+                                  main header {
+                                    width: min(100%, 18em);
+                                  }            
+                                  main header svg {
+                                    height: auto;
+                                    max-width: 100%;
+                                    width: 100%;
+                                  }            
+                                  main article {
+                                    margin-top: -0.95em;
+                                    width: min(100%, 30em);
+                                  }            
+                                  main article p {
+                                    font-size: 75%;
+                                  }            
+                                  main article br {            
+                                    display: none;            
+                                    @media(min-width: 48em) {
+                                      display: inline;
+                                    }            
+                                  }          
+            """;
+
+    private final MessageSource messageSource;
+    private final LocaleResolver<HttpRequest<?>> localeResolver;
+    private final Map<LocaleStatus, String> cache = new ConcurrentHashMap<>();
+
+    DefaultHtmlErrorResponseBodyProvider(MessageSource messageSource,
+                                         LocaleResolver<HttpRequest<?>> localeResolver) {
+        this.messageSource = messageSource;
+        this.localeResolver = localeResolver;
+    }
+
+    @Override
+    public String body(@NonNull ErrorContext errorContext, @NonNull HttpResponse<?> response) {
+        final HttpStatus status = response.status();
```

Comment:
this can error for custom status codes

Reply:
I don't find any reference in the documentation about custom status codes? https://docs.micronaut.io/latest/guide/#statusAnnotation

where is that?

Reply:
I assume is this is the PR for custom HTTP Status https://github.com/micronaut-projects/micronaut-core/pull/8147/files

Reply:
Is this what you are suggesting? https://github.com/micronaut-projects/micronaut-core/pull/11210/commits/27a15c35d0938d25fd9fd65231b823cab430839c 

do you suggest to use `HttpResponse::code` instead of `HttpResponse::status`?

Reply:
yep use .code and .reason. Note however that .reason is potentially user-controlled.